modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/programs/Download.scala

```diff
@@ -1,30 +1,51 @@
 package org.tessellation.sdk.infrastructure.snapshot.programs
 
+import cats.Monad
 import cats.effect.Async
 import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import scala.concurrent.duration.{Duration, DurationInt}
 
 import org.tessellation.schema.node.NodeState
+import org.tessellation.schema.peer.Peer
 import org.tessellation.sdk.domain.node.NodeStorage
 import org.tessellation.sdk.infrastructure.snapshot.SnapshotConsensus
 
 object Download {
 
   def make[F[_]: Async](
     nodeStorage: NodeStorage[F],
-    consensus: SnapshotConsensus[F, _, _, _, _, _]
+    consensus: SnapshotConsensus[F, _, _, _, _, _],
+    getPeers: () => F[Set[Peer]]
   ): Download[F] =
     new Download[F](
       nodeStorage,
-      consensus
+      consensus,
+      getPeers
     ) {}
 }
 
 sealed abstract class Download[F[_]: Async] private (
   nodeStorage: NodeStorage[F],
-  consensus: SnapshotConsensus[F, _, _, _, _, _]
+  consensus: SnapshotConsensus[F, _, _, _, _, _],
+  getPeers: () => F[Set[Peer]]
 ) {
+  private val timeout: Duration = 15.seconds
+  private val minKnownPeers = 50
+  private val sleepInterval = 1.second
```

Comment:
can you use more descriptive names? like: `delayBeforeDownload` instead of `sleepInterval` or something like that. as `sleepInterval` indeed says what it is but does not say why/for what. (see: imperative vs declarative programming)

Reply:
Changed.